tools/cli/helpers/dependencyAnalysis.js

```diff
@@ -186,3 +186,44 @@ export function getBuildOrder( deps ) {
 
 	return ret;
 }
+
+/**
+ * Get the max "depth" of all projects from a set of targets.
+ *
+ * If there are cycles, the "depth" for affected projects will be `deps.size + 1`.
+ * Any projects not reachable from any target will have a depth of Infinity.
+ *
+ * @param {Map}      deps    - Project dependency map.
+ * @param {Iterable} targets - Targets.
+ * @return {Map<string,number>} - Map of projects to depths.
+ */
+export function getDependencyDepths( deps, targets ) {
+	const depths = new Map();
+
+	for ( const p of deps.keys() ) {
+		depths.set( p, Infinity );
+	}
+
+	const maxDepth = deps.size + 1;
+	const queue = [];
+	for ( const p of targets ) {
+		if ( deps.has( p ) ) {
+			depths.set( p, 0 );
+			queue.push( p );
+		}
+	}
+
+	while ( queue.length > 0 ) {
+		const p = queue.shift();
+		const depth = Math.min( depths.get( p ) + 1, maxDepth );
+		for ( const p2 of deps.get( p ) ) {
+			const d2 = depths.get( p2 );
+			if ( d2 < depth || d2 === Infinity ) {
```

Comment:
I've convinced myself both ways here, but is this backwards? Should it be `>`?

```suggestion
			if ( d2 > depth || d2 === Infinity ) {
```

It might be what is desired, but I think it also hides the fact that there is a shorter depth available.

Reply:
The function is documented as finding the maximum depth (ignoring cycles). Which is what we'll probably want for the build prioritizing experiment, prioritize builds higher when there's a longer chain of stuff depending on them.